pkg/providers/kube/client_test.go

```diff
@@ -750,3 +753,140 @@ func TestListEndpointsForIdentity(t *testing.T) {
+			"7.7.7.7:888",
+		},
+		// Global should have 1 or more endpoints
+		"bookwarehouse.default.svc.cluster.global": {
+			"7.7.7.7:888",
+		},
+		// Global should have 1 or more endpoints
+		"bookwarehouse.default.svc.cluster.globalB": {
+			"7.7.7.7:888", "8.8.8.8:999",
+		},
+		// Non-Local and Non-Global should have 1 or more endpoint
+		"bookwarehouse.default.svc": {
+			"7.7.7.7:888",
+		},
+		// Non-Local and Non-Global should have 1 or more endpoint
+		"bookwarehouse.default.svcB": {
+			"7.7.7.7:888", "8.8.8.8:999",
+		},
+	}
+	// Mocked GetMultiClusterService() func
+	GetMultiClusterService := func(fqdn string) []endpoint.Endpoint {
+		var endpoints []endpoint.Endpoint
+		for _, v := range actualClusterEndpoints[fqdn] {
+			s := strings.Split(v, ":")
+			port, _ := strconv.ParseUint(s[1], 10, 32)
+			ept := endpoint.Endpoint{
+				IP:   []byte(s[0]),
+				Port: endpoint.Port(port),
+			}
+			// Keep appending as there are potentially multiple endpoints
+			endpoints = append(endpoints, ept)
+		}
+		return endpoints
+	}
+
+	testCases := []struct {
+		name              string
+		isLocal           bool
+		isGlobal          bool
+		svcFqdn           string
+		expectedEndpoints []endpoint.Endpoint
+	}{
+		{
+			name:     "Local should have 1 endpoint",
+			isLocal:  true,
+			isGlobal: false,
+			svcFqdn:  "bookwarehouse.default.svc.cluster.local",
+			expectedEndpoints: []endpoint.Endpoint{
+				{
+					IP:   net.ParseIP("7.7.7.7"),
+					Port: 888,
+				},
+			},
+		},
+		{
+			name:     "Global should have 1 more more endpoints",
+			isLocal:  false,
+			isGlobal: true,
+			svcFqdn:  "bookwarehouse.default.svc.cluster.global",
+			expectedEndpoints: []endpoint.Endpoint{
+				{
+					IP:   net.ParseIP("7.7.7.7"),
+					Port: 888,
+				},
+			},
+		},
+		{
+			name:     "Global should have 1 more more endpoints",
+			isLocal:  false,
+			isGlobal: true,
+			svcFqdn:  "bookwarehouse.default.svc.cluster.globalB",
+			expectedEndpoints: []endpoint.Endpoint{
+				{
+					IP:   net.ParseIP("7.7.7.7"),
+					Port: 888,
+				},
+				{
+					IP:   net.ParseIP("8.8.8.8"),
+					Port: 999,
+				},
+			},
+		},
+		{
+			name:     "Non-Local and Non-Global hould have 1 or more endpoints",
+			isLocal:  false,
+			isGlobal: false,
+			svcFqdn:  "bookwarehouse.default.svc",
+			expectedEndpoints: []endpoint.Endpoint{
+				{
+					IP:   net.ParseIP("7.7.7.7"),
+					Port: 888,
+				},
+			},
+		},
+		{
+			name:     "Non-Local and Non-Global should have 1 or more endpoints",
+			isLocal:  false,
+			isGlobal: false,
+			svcFqdn:  "bookwarehouse.default.svcB",
+			expectedEndpoints: []endpoint.Endpoint{
+				{
+					IP:   net.ParseIP("7.7.7.7"),
+					Port: 888,
+				},
+				{
+					IP:   net.ParseIP("8.8.8.8"),
+					Port: 999,
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.isLocal && !tc.isGlobal {
+				// Local should have just one endpoint
+				endpoints := GetMultiClusterService(tc.svcFqdn)
```

Comment:
so we'll want to actually call the real function, otherwise the value of the tests are severely diminished.

This means you'll have to add a mock configClient and a mock kubeController